File: app/src/main/java/id/my/hiemz/wms/TeksAdapter.java
package id.my.hiemz.wms;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;

// Adapter teks sederhana (dipakai layar riwayat).
public class TeksAdapter extends RecyclerView.Adapter<TeksAdapter.VH> {
    private ArrayList<String> data;
    public TeksAdapter(ArrayList<String> d) { data = d; }
    static class VH extends RecyclerView.ViewHolder {
        TextView t;
        VH(View v) { super(v); t = v.findViewById(R.id.rowTeks); }
    }
    @Override public VH onCreateViewHolder(ViewGroup p, int t) {
        return new VH(LayoutInflater.from(p.getContext()).inflate(R.layout.row_teks, p, false));
    }
    @Override public void onBindViewHolder(VH h, int i) { h.t.setText(data.get(i)); }
    @Override public int getItemCount() { return data.size(); }
}

File: app/src/main/java/id/my/hiemz/wms/TeksAdapter.kt
package id.my.hiemz.wms

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import id.my.hiemz.wms.databinding.RowTeksBinding

// Adapter teks sederhana (layar riwayat).
class TeksAdapter(private val data: ArrayList<String>) : RecyclerView.Adapter<TeksAdapter.VH>() {
    class VH(val b: RowTeksBinding) : RecyclerView.ViewHolder(b.root)

    override fun onCreateViewHolder(p: ViewGroup, t: Int): VH =
        VH(RowTeksBinding.inflate(LayoutInflater.from(p.context), p, false))

    override fun onBindViewHolder(h: VH, i: Int) { h.b.rowTeks.text = data[i] }
    override fun getItemCount() = data.size
}
